addons/l10n_es_edi_facturae/models/account_edi_format.py

```diff
@@ -0,0 +1,499 @@
+            raise UserError(_('The company needs a set tax identification number or VAT number'))
+
+        if not partner.l10n_es_edi_facturae_tax_identifier:
+            raise UserError(_('The partner needs a set tax identification number or VAT number'))
+
+        def fmt_eur(value):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.850...1).
+            return float_repr(value, eur_curr.decimal_places)
+
+        def fmt_curr(number):
+            # Format the monetary values to avoid trailing decimals (e.g. 90.850...1).
+            if inv_curr.name == 'EUR':
+                return fmt_eur(number)
+            return float_repr(number, inv_curr.decimal_places)
+
+        eur_curr = self.env['res.currency'].search([('name', '=', 'EUR')])
+        inv_curr = invoice.currency_id or self_company.currency_id
+        IndividualName = namedtuple('IndividualName', ['firstname', 'surname', 'surname2'])
+        firstname = surname = surname2 = ''
+        phone_clean_table = str.maketrans({" ": None, "-": None, "(": None, ")": None, "+": None})
+        legal_literals = invoice.narration.striptags() if invoice.narration else False
+        legal_literals = legal_literals.split(";") if legal_literals else False
+
+        if partner.is_company:
+            partner_name = IndividualName('', '', '')
+        else:
+            name_split = [part for part in partner.name.replace(', ', ' ').split(' ') if part]
+            if len(name_split) > 2:
+                firstname = ' '.join(name_split[:-2])
+                surname, surname2 = name_split[-2:]
+            elif len(name_split) == 2:
+                firstname = ' '.join(name_split[:-1])
+                surname = name_split[-1]
+
+            partner_name = IndividualName(firstname, surname, surname2)
+
+        template_values = EDI_XML_DICT_TEMPLATES['facturae'].copy()
+        self_party_values = EDI_XML_DICT_TEMPLATES['business'].copy()
+        partner_party_values = EDI_XML_DICT_TEMPLATES['business'].copy()
+        signature_values = EDI_XML_DICT_TEMPLATES['signature'].copy()
+
+        if invoice.move_type == "entry":
+            return False
+        invoice_issuer_type = 'EM' if invoice.move_type == 'out_invoice' else 'RE'
+        invoice_issuer_signature_type = 'supplier' if invoice.move_type == 'out_invoice' else 'customer'
+
+        company_is_not_euro = self_company.currency_id != eur_curr
+        invoice_is_not_euro = inv_curr != eur_curr
+
+        if not (company_is_not_euro or invoice_is_not_euro):
+            conv_rate = 1.00
+            need_conv = False
+        elif not company_is_not_euro and invoice_is_not_euro:
+            conv_rate = inv_curr._get_conversion_rate(inv_curr, eur_curr, self_company, invoice.invoice_date)
+            need_conv = True
+        elif company_is_not_euro and invoice_is_not_euro:
+            conv_rate = inv_curr._get_conversion_rate(inv_curr, eur_curr, self_company, invoice.invoice_date)
```

Comment:
never use the conversion rate in odoo... Use the invoice's rate instead

Reply:
But the invoice line rate is just an extra step to do the exact same thing